src/debugAdapter/goDebug.ts

```diff
@@ -900,12 +946,21 @@ class GoDebugSession extends LoggingDebugSession {
 		});
 	}
 
-	protected setBreakPointsRequest(response: DebugProtocol.SetBreakpointsResponse, args: DebugProtocol.SetBreakpointsArguments): void {
+	protected async setBreakPointsRequest(response: DebugProtocol.SetBreakpointsResponse, args: DebugProtocol.SetBreakpointsArguments): Promise<void> {
 		log('SetBreakPointsRequest');
-		if (!this.continueRequestRunning) {
-			this.setBreakPoints(response, args);
+		try {
+			// If a program is launched with --continue, the program is running
+			// before we can run attach. So we would need to check the state.
+			// We use NonBlocking so the call would return immediately.
+			this.debugState = await this.delve.getDebugState();
+		} catch (error) {
+			logError(`Failed to get state ${String(error)}`);
+		}
+
+		if (!this.debugState.Running && !this.continueRequestRunning) {
```

Comment:
If you failed to get the state in the above step, then wont debugState be null/undefined?